driver/registry_default_registration.go

```diff
@@ -8,17 +8,23 @@ import (
 	"github.com/ory/kratos/selfservice/flow/registration"
 )
 
+func filter(hooks []config.SelfServiceHook, persistencePhase string) (ret []config.SelfServiceHook) {
+	for _, h := range hooks {
+		if len(h.PersistencePhase) == 0 || // empty value means default = "all"
+			h.PersistencePhase == "all" || // default value
+			h.PersistencePhase == persistencePhase {
```

Comment:
I think it would be a good idea to also add an enum check in golang  not only in jsonnet